inst/include/RcppSimdJson/deserialize.hpp

```diff
@@ -466,25 +468,24 @@ inline SEXP query_and_deserialize(simdjson::dom::element                       p
     }
 
     if constexpr (query_error_ok) {
-        simdjson::dom::element queried;
-        if(parsed.at_pointer(std::string_view(query)).get(queried) == simdjson::SUCCESS) {
-            return deserialize(queried, parse_opts);				// #nocov
+        auto queried = parsed.at_pointer(std::string_view(query));
+        if(queried.second == simdjson::SUCCESS) {
```

Comment:
This troubles me that you are able to do queried.second. I thought we made this impossible..

The syntax is supposed to be...
```C++
ondemand::value queried;
simdjson::error_code error = parsed.at_pointer(std::string_view(query)).get(queried);
if(queried ==...}
```

Does that no work?

Reply:
That is what I had in the before, but if you go back to yesterday's commit e91ee40, I still have an error with regard to the `get()` method which says that it is not implemented with the given type (`simdjson::ondemand::value`). And also, when I look in the documentation, it says that is should not be supported, so I was surprised to see it work in our own tests.

Reply:
I'll investigate right now. Meanwhile please don't use first/second. We don't want end users to use it. It is confusing and error prone.

Reply:
Do you want me to revert to commit e91ee40 right now?

Reply:
I just finished producing a PR directly on simdjson so that it will no longer possible to use first/second. I don't have yet a good idea of the issue you are encountering. Let have a look first.

Reply:
I'll need more coffee too!

Reply:
Ok I will revert then, so this is not broken code.